test/es6/proxybugs.js

```diff
@@ -383,6 +383,49 @@ var tests = [
             assert.isTrue(trapCalled);
         }
     },
+    {
+        name: "Extending a proxy with an es6 class",
+        body() {
+            function parent() { this.noTrap = true; }
+            var proxyNoTrap = new Proxy(parent, {});
+            var handler = {
+                get construct () {
+                    this.trap = true;
+                }
+            }
+            var proxyWithTrap = new Proxy(parent, handler );
+
+            class NoTrap extends proxyNoTrap
+            {
+                constructor()
+                {
+                    super();
+                    this.own = true;
+                }
+                a () { return true; }
+            }
+
+            class WithTrap extends proxyWithTrap
+            {
+                constructor()
+                {
+                    super();
+                    this.own = true;
+                }
+                a () { return true; }
+            }
+            
+            var notrap = new NoTrap();
+            assert.isTrue(notrap.own);
+            assert.isTrue(notrap.a());
+            assert.isTrue(notrap.noTrap);
+            
+            var withtrap = new WithTrap();
+            assert.isTrue(withtrap.own);
+            assert.isTrue(withtrap.a());
+            assert.isUndefined(withtrap.trap);

```

Comment:
Shouldn't this be `withtrap.noTrap`?

Reply:
withtrap.trap was intentional - it's undefined as the "this" in the constructor trap is not the same this as the one for the target object.

I could add withtrap.noTrap as well, and I suppose I could make the trap do something to prove that the trap is working.

Key intent here was testing that the class extension is working correctly without picking up extra "this"s.

Reply:
Oh, I misread what the test was doing. I thought you were verifying that using the proxy with the construct trap meant that it didn't actually call the original constructor function. I'm not sure I understand what you are trying to verify here; doesn't the `this` given to the proxy construct function get returned as the constructed object, so the `this.trap = true` should set the property on the instance?

Reply:
I've expanded the case a bit further. I note that it covers more than it needs to - there were no tests of any kind (that I could see) for class that extend proxies so I thought I should do more than the specific point this patch fixes.